.github/workflows/code-hygiene.yml

```diff
@@ -83,3 +83,17 @@ jobs:
           fi
         done
         exit $exit_code
+
+  check-println:
```

Comment:
Instead of a new check, what about leveraging checkstyle's RegexpSingleline, below was how the 'extension' filter was added, maybe we could add entries for system.out.print & system.err.print that would be easier to maintain

```xml
  <module name="RegexpSingleline">
    <property name="format" value="extension"/>
    <property name="ignoreCase" value="true"/>
    <property name="message" value="Extension should only be used sparingly to keep implementations as generic as possible" />
    <property name="severity" value="error"/>
  </module>
```

Reply:
I tried that but didn't have a working solution, I'll invest some more time to see if I can make it work

Reply:
I was not able to have it running with:
```xml
  <module name="RegexpSingleline">
    <property name="format" value="System\.out\.println" />
    <property name="message" value="Do not use System.out.println" />
    <!-- Exception list of files and directories -->
    <property name="fileExtensions" value="java" />
    <property name="exclude" value="./src/main/java/org/opensearch/security/tools, ./src/main/java/com/amazon/dlic/auth/http/kerberos/HTTPSpnegoAuthenticator.java" />
  </module>
```

Reply:
Try building off of this, works locally for me
```xml
  <module name="RegexpSingleline">
    <property name="format" value="println"/>
    <property name="ignoreCase" value="true"/>
    <property name="message" value="SYSTEM.OUT.PRINTLN NONONO!" />
    <property name="severity" value="error"/>
  </module>
```

```
./gradlew checkstyleMain 
...
> Task :checkstyleMain
[ant:checkstyle] [ERROR] /local/home/petern/git/security/src/main/java/org/opensearch/security/securityconf/DynamicConfigFactory.java:247: SYSTEM.OUT.PRINTLN NONONO! [RegexpSingleline]
[ant:checkstyle] [ERROR] /local/home/petern/git/security/src/main/java/org/opensearch/security/securityconf/DynamicConfigFactory.java:248: SYSTEM.OUT.PRINTLN NONONO! [RegexpSingleline]
[ant:checkstyle] [ERROR] /local/home/petern/git/security/src/main/java/org/opensearch/security/auditlog/sink/AuditLogSink.java:63: SYSTEM.OUT.PRINTLN NONONO! [RegexpSingleline]
[ant:checkstyle] [ERROR] /local/home/petern/git/security/src/main/java/org/opensearch/security/auditlog/sink/DebugSink.java:30: SYSTEM.OUT.PRINTLN NONONO! [RegexpSingleline]
[ant:checkstyle] [ERROR] /local/home/petern/git/security/src/main/java/org/opensearch/security/ssl/OpenSearchSecuritySSLPlugin.java:161: SYSTEM.OUT.PRINTLN NONONO! [RegexpSingleline]
[ant:checkstyle] [ERROR] /local/home/petern/git/security/src/main/java/org/opensearch/security/ssl/OpenSearchSecuritySSLPlugin.java:162: SYSTEM.OUT.PRINTLN NONONO! [RegexpSingleline]
[ant:checkstyle] [ERROR] /local/home/petern/git/security/src/main/java/org/opensearch/security/ssl/OpenSearchSecuritySSLPlugin.java:228: SYSTEM.OUT.PRINTLN NONONO! [RegexpSingleline]
[ant:checkstyle] [ERROR] /local/home/petern/git/security/src/main/java/org/opensearch/security/ssl/OpenSearchSecuritySSLPlugin.java:229: SYSTEM.OUT.PRINTLN NONONO! [RegexpSingleline]
[ant:checkstyle] [ERROR] /local/home/petern/git/security/src/main/java/org/opensearch/security/tools/AuditConfigMigrater.java:94: SYSTEM.OUT.PRINTLN NONONO! [RegexpSingleline]
[ant:checkstyle] [ERROR] /local/home/petern/git/security/src/main/java/org/opensearch/security/tools/AuditConfigMigrater.java:109: SYSTEM.OUT.PRINTLN NONONO! [RegexpSingleline]
[ant:checkstyle] [ERROR] /local/home/petern/git/security/src/main/java/org/opensearch/security/tools/AuditConfigMigrater.java:112: SYSTEM.OUT.PRINTLN NONONO! [RegexpSingleline]
```

Reply:
Yes, I the problem is how to exclude. There is no such Property called `exclude`. Do you know of anything of that might work?
Closest thing I found was this: https://github.com/checkstyle/checkstyle/blob/master/src/main/resources/sun_checks.xml#L46-L50
However this applies to whole file, so I might have to end up creating a separate xml for sysout check anyway.

Reply:
I was able to achieve this by adding a separate file for println checks.

Reply:
That too hard - exclusions can be done in place, see https://github.com/opensearch-project/security/blob/main/DEVELOPER_GUIDE.md#checkstyle-violations